cmd/docker-mcp/internal/gateway/handlers.go

```diff
@@ -2,16 +2,41 @@ package gateway
 
 import (
 	"context"
+	"fmt"
+	"os"
+	"time"
 
 	"github.com/modelcontextprotocol/go-sdk/mcp"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/metric"
 
 	"github.com/docker/mcp-gateway/cmd/docker-mcp/internal/catalog"
+	"github.com/docker/mcp-gateway/cmd/docker-mcp/internal/telemetry"
 )
 
 func getClientConfig(readOnlyHint *bool, ss *mcp.ServerSession, server *mcp.Server) *clientConfig {
 	return &clientConfig{readOnly: readOnlyHint, serverSession: ss, server: server}
 }
 
+// inferServerType determines the type of MCP server based on its configuration
+func inferServerType(serverConfig *catalog.ServerConfig) string {
+	// Check for SSE endpoint first (including remote URLs)
+	if serverConfig.Spec.SSEEndpoint != "" || serverConfig.Spec.Remote.URL != "" {
+		return "sse"
+	}
+	// Check for Docker image
+	if serverConfig.Spec.Image != "" {
+		return "docker"
+	}
+	// Check for stdio command
+	if len(serverConfig.Spec.Command) > 0 {
+		return "stdio"
```

Comment:
See above.  I'm not sure what this really tells us today.  Docker and stdio are the same thing.